### ISSUE TITLE
bug(cli): need to print all the diagnostics and exit with 1 in CI

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#1403]
The current implementation can deny warnings by `oxlint --max-warnings 0`, but then it'll stop printing anything to the terminal.

ESLint has to manually configure them in a config file, which is against our zero-config philosophy.

Rust has the `-D warnings` flag for this use case in CIs. I think this should be the right approach for this problem.

So to print all the diagnostics and exit with 1 in CI environments, we should go for

```
oxlint --deny-warnings
```

**Comment by hyf0** [comment]
Should also consider the priority.

For `oxlint --deny-warnings --allow-some-warning-rule`, should not throw warnings for `some-warning-rule`.
